nav2_collision_monitor/src/kinematics.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_collision_monitor/kinematics.hpp"
+
+#include <cmath>
+
+namespace nav2_collision_monitor
+{
+
+void transformPoint(const Velocity & velocity, const double dt, Point & point)
+{
+  // p = R*p' + vel*dt
+  // p' = Rt * (p - vel*dt)
+  // R = [ cos_theta sin_theta]
+  //     [-sin_theta cos_theta]
+
+  const double mul_x = point.x - velocity.x * dt;
+  const double mul_y = point.y - velocity.y * dt;
+  const double theta = velocity.tw * dt;
+  const double cos_theta = std::cos(theta);
+  const double sin_theta = std::sin(theta);
+  point.x = mul_x * cos_theta + mul_y * sin_theta;
+  point.y = -mul_x * sin_theta + mul_y * cos_theta;
+}
+
+void transformPoint(const Pose & curr_pose, Point & point)
+{
+  // p = R*p' + curr_pose
+  // p' = Rt * (p - curr_pose)
+  const double mul_x = point.x - curr_pose.x;
+  const double mul_y = point.y - curr_pose.y;
+  const double cos_theta = std::cos(curr_pose.theta);
+  const double sin_theta = std::sin(curr_pose.theta);
+  point.x = mul_x * cos_theta + mul_y * sin_theta;
+  point.y = -mul_x * sin_theta + mul_y * cos_theta;
+}
+
+void stepRobot(Velocity & velocity, const double dt, Pose & pose)
```

Comment:
I don't understand `stepRobot`

Reply:
This is projecting forward by a kinematic model (though the order of arguments is confusing; put `dt` either first or last as the only `const` not being modified in the function.

`projectState` perhaps